src/python/pants/backend/python/subsystems/ipython.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from typing import List
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+
+
+class IPython(PythonToolBase):
+  options_scope = 'ipython'
+  default_version = 'ipython==5.8.0'
+  default_extra_requirements: List[str] = []
+  default_entry_point = 'IPython:start_ipython'
+  default_interpreter_constraints = ["CPython>=2.7,<3", "CPython>=3.4"]
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register(
+      '--enabled', type=bool, default=False, fingerprint=True,
```

Comment:
@benjyw discussed what this option should look like in a V2 world. Originally, we had `--repl-py-ipython` as a bool option that would enable IPython. 

We at first discussed a goal-level option `--repl-ipython`, but that doesn't make sense in the context of JVM, for example. Nor could we come up with a generic option like we have with `--test-coverage`.

Instead, we settled on `--ipython-enabled`. Why not `--ipython-skip`, which we use with linters/formatters? Unlike `lint`, we don't expect to run _multiple REPL tools_ over the same Python target. This comes down to whether you want the standard REPL or the souped-up IPython REPL. Skipping is orthogonal.